easypost/user.py

```diff
@@ -1,66 +1,62 @@
-        return convert_to_easypost_object(response, api_key)
+        response, api_key = requestor.request(method="post", url=url, params=wrapped_params, api_key_required=False)
+        return convert_to_easypost_object(response=response, api_key=api_key)
 
     @classmethod
-    def retrieve(cls, easypost_id="", api_key=None, **params):
+    def retrieve(cls, easypost_id: Optional[str] = None, api_key: Optional[str] = None, **params) -> "User":
         """Retrieve a user."""
-        try:
-            easypost_id = easypost_id["id"]
-        except (KeyError, TypeError):
-            pass
-
-        if easypost_id == "":
-            requestor = Requestor(api_key)
-            response, api_key = requestor.request("get", cls.class_url())
-            return convert_to_easypost_object(response, api_key)
+        if not easypost_id:
+            requestor = Requestor(local_api_key=api_key)
+            response, api_key = requestor.request(method="get", url=cls.class_url())
+            return convert_to_easypost_object(response=response, api_key=api_key)
         else:
-            instance = cls(easypost_id, api_key, **params)
+            instance = cls(easypost_id=easypost_id, api_key=api_key, **params)
             instance.refresh()
             return instance
 
     @classmethod
-    def retrieve_me(cls, api_key=None, **params):
+    def retrieve_me(cls, api_key: Optional[str] = None, **params) -> "User":
         """Retrieve the authenticated user."""
-        requestor = Requestor(api_key)
-        response, api_key = requestor.request("get", cls.class_url())
-        return convert_to_easypost_object(response, api_key)
+        requestor = Requestor(local_api_key=api_key)
+        response, api_key = requestor.request(method="get", url=cls.class_url())
+        return convert_to_easypost_object(response=response, api_key=api_key)
 
     @classmethod
-    def all_api_keys(cls, api_key=None):
+    def all_api_keys(cls, api_key: Optional[str] = None) -> "User":
         """Get all API keys including child user keys."""
-        requestor = Requestor(api_key)
+        requestor = Requestor(local_api_key=api_key)
         url = "/api_keys"
-        response, api_key = requestor.request("get", url)
-        return convert_to_easypost_object(response, api_key)
+        response, api_key = requestor.request(method="get", url=url)
+        return convert_to_easypost_object(response=response, api_key=api_key)
 
-    def api_keys(self):
+    def api_keys(self) -> List[str]:
         """Get the authenticated user's API keys."""
         api_keys = self.all_api_keys()
 
         if api_keys.id == self.id:
-            my_api_keys = api_keys.keys
+            return api_keys.keys
         else:
             for child in api_keys.children:
                 if child.id == self.id:
-                    my_api_keys = child.keys
-                    break
+                    return child.keys
 
-        return my_api_keys
+        return []
```

Comment:
Why is this here? There shouldn't ever be a time where we return an empty list as either the request itself will fail or a list of API keys will get returned. You can't make a request to get API keys with an API key and not get one back.

Reply:
Then it should never reach this line 😉 , but if it ever does.... but it should never, but just in case...

Reply:
It can literally never get here haha. Code that serves no purpose shouldn't be in the repo.

Reply:
Watch. One day something catastrophic will happen that will trigger this, and rather than the user getting nothing back from this function, and attempting to use a `None` later that will blow up their code stack, they'll get just a friendly empty list instead.

Reply:
In any other programming language, a function written without this would complain that not all routes return a value, and may not even compile. Just because Python doesn't care doesn't mean we shouldn't.

Reply:
See, `mypy` knows :)
<img width="547" alt="image" src="https://user-images.githubusercontent.com/17054780/160008916-c3eb6813-0ea8-48f8-962e-8422a4fe4d2f.png">